src/asm/symbol.c

```diff
@@ -276,6 +276,13 @@ static bool isReferenced(struct Symbol const *sym)
 	return sym->ID != (uint32_t)-1;
 }
 
+struct SymbolListNode {
+	struct Symbol *sym;
+	struct SymbolListNode *next;
+};
+
+static struct SymbolListNode *purgeList = NULL;
```

Comment:
I prefer avoiding global variables, as they are Not Great for various reasons. (Primarily, they require a lot more work when trying to turn something unique into several instances thereof.)

Unfortunately, the way to fix this is to make the `purge_list` in the grammar an array of symbols[^cxx], and delete each of them at the end of the line.
Why unfortunately? Because that requires making deeper changes to the grammar file—none difficult, thankfully, but if you're not used to it, it'd require some learning...

I'm willing to provide guidance/mentoring to avoid you having to read the entire GNU Bison manual, if you want to.

[^cxx]: \*grumbles\* *God*, give me C++ `std::vector` (or `Vec` lol) any time over our 13th manually-implemented specialisation of this goddamn data type...

Reply:
Oh neat, GFM has [footnotes](https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/).